frappe/public/js/frappe/desk.js

```diff
@@ -26,14 +26,19 @@ $(document).ready(function() {
 
 frappe.Application = Class.extend({
 	init: function() {
-		this.startup();
+		try {
+			this.startup();
+		} catch (e) {
+			console.log(e);
+			frappe.msgprint(__('Init ERPNext Failed: ' + (e.message || e)));
+		}
```

Comment:
Due to stuff being async in js, this will catch more stuff

```suggestion
window.addEventListener("error", function (e) {
    console.error('Unhandled error', event);
    frappe.msgprint(__('Init ERPNext Failed: ' + (e.message || e)));
})
window.addEventListener('unhandledrejection', function(event) {
    console.error('Unhandled promise rejection', event);
    frappe.msgprint(__('Init ERPNext Failed: ' + (e.message || e)));
});
this.startup();
```
```